lib/presentation/blocs/activity_cubit.dart

```diff
@@ -103,6 +105,16 @@ class ActivityCubit extends AbstractLocationCubit<ActivityState> {
   /// stream.
   void listenToLocation() {
     emit(ActivityState.loading(lastKnownLocation: _lastKnownLocation));
+
+    // First try to get the last known location, generally faster.
+    _getLastKnownLocation().then((result) {
```

Comment:
I'm not sure if we want to get the last location in the cubit.
This might cause the app to display a completely wrong location to the user:
Let's say I last used the app in Madrid and now I'm in paris, when I start the app it will clearly indicate that I am in Madrid, do we really want this behavior ?

Reply:
It's the behaviour of google maps, if you used it in madrid and launch it once you're in Paris you'll have a gray user marker in madrid before having the blue one in Paris.

By doing that we don't have a different color but we still have the loading icon so it's fine to me.

I just have to add a condition to push a state only if the state is still loading, we don't want to push a loading state if the stream has already pushed a loaded state, this seems unlikely but still